brian2/synapses/synapses.py

```diff
@@ -1570,3 +1575,17 @@ def _add_synapses_generator(self, j, n, skip_if_invalid=False, namespace=None, l
                                         check_units=False,
                                         run_namespace=namespace)
         codeobj()
+
+    def _check_synapse_parse(self, parsed, namespace):
```

Comment:
The idea would be to extend this function if any other type-checking is needed in the future. Maybe this is a bit too elaborate?